lib/internal/streams/operators.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const { AbortController } = require('internal/abort_controller');
+const { AbortError } = require('internal/errors');
+const Readable = require('internal/streams/readable');
+const eos = require('internal/streams/end-of-stream');
+
+module.exports.map = function map(stream, fn, options) {
+  // TODO: Argument validation
+
+  let concurrency = 1;
+  if (Number.isFinite(options)) {
+    concurrency = options;
+  } else if (options && Number.isFinite(options.concurrency)) {
+    concurrency = options.concurrency;
+  }
+
+  const ac = new AbortController();
+  const signal = ac.signal;
+  const queue = [];
+
+  let reading = false;
+
+  // TODO: What about hwm? This will cause some unecessary buffering.
+  const ret = new Readable({
+    objectMode: stream.readableObjectMode ?? stream.objectMode ?? true,
+    read () {
+      if (!reading) {
+        read();
+      }
+    },
+    destroy (err, callback) {
+      if (!err && !this.readableEnded) {
+        err = new AbortError();
+      }
```

Comment:
```suggestion
      }
      stream.destroy(err);
```

Should failure back propagate to source stream?

Reply:
@devsnek does `.return` propagate in proposal-iterator-helpers? 

That is, in that proposal:

```js
const iter = getIterator();
const mapped = iter.map(x => x * 2);
mapped.return(); // is `.return` called on iter?
```

Reply:
@ronag did you resolve because you checked with that? Because I wanna make sure we align

Reply:
I resolved it because I added a destroy. It was before I saw your comment.